src/anndata/_io/specs/methods.py

```diff
@@ -391,7 +390,50 @@ def write_basic(
     dataset_kwargs: Mapping[str, Any] = MappingProxyType({}),
 ):
     """Write methods which underlying library handles natively."""
-    f.create_dataset(k, data=elem, **dataset_kwargs)
+    dtype = dataset_kwargs.get("dtype", elem.dtype)
+    f.create_dataset(k, data=elem, **dataset_kwargs, dtype=dtype)
+
+
+def _iter_chunks_for_copy(elem: ArrayStorageType, dest: ArrayStorageType):
+    """
+    Returns an iterator of tuples of slices for copying chunks from `elem` to `dest`.
+
+    * If `dest` has chunks, it will return the chunks of `dest`.
+    * If `dest` is not chunked, we write it in ~100MB chunks or 1000 rows, whichever is larger.
+    """
+    if dest.chunks and hasattr(dest, "iter_chunks"):
+        return dest.iter_chunks()
+    else:
+        itemsize = elem.dtype.itemsize
+        shape = elem.shape
+        # Number of elements to write
+        entry_chunk_size = 100 * 1024 * 1024 // itemsize
+        # Number of rows that works out to
+        n_rows = max(entry_chunk_size // shape[0], 1000)
```

Comment:
should any of this be configurable?

Reply:
The default as things stand is currently dependent on both the shape and an arbitrary cutoff in `max`....so Given the current implementation, we could make 2-3 things configurable which seems like overkill.  Perhaps just `n_rows` should be a setting with `1000` as the default?

Reply:
We already have https://github.com/scverse/anndata/blob/df213f659f0e9eadfcab4af48ee98de7145252a7/src/anndata/_io/h5ad.py#L176, documented as “Used only when loading sparse dataset that is stored as dense.”

Also https://github.com/scverse/anndata/blob/df213f659f0e9eadfcab4af48ee98de7145252a7/src/anndata/_io/zarr.py#L30

Let’s not create multiple different conventions under the same name.

Reply:
So this is an argument for not calling it `chunk_size`? I wasn't proposing literally calling it `n_rows` but just that variable being the settings as opposed to `entry_chunk_size` or the `max` value

Reply:
It’s an argument for keeping our terminology consistent.